mycroft_bus_client/client/client.py

```diff
@@ -198,6 +304,56 @@ def emit(self, message):
             LOG.warning('Could not send %s message because connection '
                         'has been closed', message.msg_type)
 
+    def collect_responses(self, message,
+                          min_timeout=0.2, max_timeout=3.0,
+                          direct_return_func=lambda msg: False):
+        """Collect responses from multiple handlers.
+
+        This sets up a collect-call (pun intended) expecting multiple handlers
+        to respond.
+
+        Args:
+            message (Message): message to send
+            min_timeout (int/float): Minimum time to wait for a response
+            max_timeout (int/float): Maximum allowed time to wait for an answer
```

Comment:
I like the initial ack response, is that used elsewhere already? 

So the min_timeout is to collect ack's, then it would wait up to the max_timeout for success/failure?

Does this remove the need for requesting an extension? I guess the extension says "I promise I'm still trying", but if something says it will provide a response and then fails to respond, that's a bug we should be catching.

Reply:
The ack is probably a new concept.

As mentioned above I the request extension could be useful in case a handler has hung and isn't responding. It adds much complexity and the gain is little so I'm not convinced we should have it.

I can definitely add a log "These said they'd handle the request but did not reply within the timeout"